velox/type/Type.h

```diff
@@ -435,6 +435,37 @@ struct TypeFactory;
     return this->kind() == TypeKind::KIND;                                \
   }
 
+class Type;
```

Comment:
We should be able to define an alias on this forward-declaration.
`using TypePtr = std::shared_ptr<const Type>;`